pkg/generators/models/models.go

```diff
@@ -144,78 +147,114 @@ func (m *Model) Render(ctx context.Context, writer io.Writer) error {
 	return nil
 }
 
-func resolveAllOf(ref *openapi3.SchemaRef) (out *openapi3.SchemaRef) {
+func resolveAllOf(ref *openapi3.SchemaRef, passed passedSchemas) (out *openapi3.SchemaRef) {
+	if ref == nil {
+		ref = &openapi3.SchemaRef{}
+	}
+	if ref.Value == nil {
+		ref.Value = &openapi3.Schema{}
+	}
+
 	out = ref
+
+	if passed == nil {
+		passed = make(passedSchemas)
+	}
+
 	for _, subSchema := range ref.Value.AllOf {
-		out = mergeSchemas(out, subSchema)
+		if _, exists := passed[subSchema]; exists {
+			continue
+		}
```

Comment:
this is what i was originally expecting to break the recursion loop

Reply:
well, I missed it in the review, didn't understand entirely it there.